sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/DateTimeUtils.scala

```diff
@@ -385,9 +385,16 @@ object DateTimeUtils {
    * microseconds where microsecond 0 is 1970-01-01 00:00:00Z.
    */
   def instantToMicros(instant: Instant): Long = {
-    val us = Math.multiplyExact(instant.getEpochSecond, MICROS_PER_SECOND)
-    val result = Math.addExact(us, NANOSECONDS.toMicros(instant.getNano))
-    result
+    val secs = instant.getEpochSecond
+    if (secs == Math.floorDiv(Long.MinValue, MICROS_PER_SECOND)) {
+      val us = Math.multiplyExact(secs + 1, MICROS_PER_SECOND)
```

Comment:
We know `sec` in advance, so, we can calculate `us` in advance too. Please, fold the arithmetic expression.

Reply:
ok thanks

Reply:
Are you sure? as us might be (secs+1) or secs depend on case

Reply:
`Math.floorDiv(Long.MinValue, MICROS_PER_SECOND)` is constant, right?

Reply:
I mostly care of the comparison:
```scala
secs == Math.floorDiv(Long.MinValue, MICROS_PER_SECOND)
```
which is performed per every input. Probably, JVM could optimize it but I would help to it and introduced a constant.

Reply:
Yea this should be a constant, but after we enter the if branch and reach here, it's rare and I'm OK to not have a constant for `val us = ...`

Reply:
Done, thanks